File: HorseRacing.Contracts/Models/Authentication/Requests/Registration/RegistrationRequest.cs
using HorseRacing.Contracts.Base.Requests;

namespace HorseRacing.Contracts.Models.Authentication.Requests.Registration
{
    public class RegistrationRequest : BaseRequest<RegistrationRequestDto>
    {
        public RegistrationRequest() : base() { }
        public RegistrationRequest(RegistrationRequestDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/auth/requests/registration/registration-request.ts
import type { BaseRequest } from '~/interfaces/api/contracts/base/requests/base-request';
import type { RegistrationRequestDto } from './registration-request-dto';

export class RegistrationRequest implements BaseRequest<RegistrationRequestDto> {
    Data: RegistrationRequestDto;

    constructor(data?: RegistrationRequestDto) {
        this.Data = data || {
            FirstName: '',
            LastName: '',
            UserName: '',
            Phone: '',
            Email: '',
            Password: '',
        };
    }
}
